File: myFinance_improved_V2/lib/features/cash_transaction/domain/entities/cash_location.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'cash_location.freezed.dart';

/// Domain entity representing a cash location
///
/// Maps to `cash_locations` table
/// DB columns: cash_location_id, location_name, location_type, store_id, company_id
@freezed
class CashLocation with _$CashLocation {
  const factory CashLocation({
    required String cashLocationId,
    required String locationName,
    required String locationType,
    String? storeId,
    String? companyId,
    String? accountId,
  }) = _CashLocation;

  const CashLocation._();

  /// Check if this is a cash drawer
  bool get isCash => locationType == 'cash';

  /// Check if this is a bank account
  bool get isBank => locationType == 'bank';

  /// Check if this is a vault/safe
  bool get isVault => locationType == 'vault';
}

File: website/src/features/cash_ending/domain/entities/CashLocation.ts
/**
 * CashLocation Entity
 * Represents a cash location (cash register, bank, vault)
 */

export type LocationType = 'cash' | 'bank' | 'vault';

export interface CashLocation {
  cash_location_id: string;
  location_name: string;
  location_type: LocationType;
  store_id: string | null;
  store_name: string | null;
  company_id: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export class CashLocationEntity {
  constructor(
    public readonly cashLocationId: string,
    public readonly locationName: string,
    public readonly locationType: LocationType,
    public readonly storeId: string | null,
    public readonly storeName: string | null,
    public readonly companyId: string,
    public readonly isActive: boolean,
    public readonly createdAt: string,
    public readonly updatedAt: string
  ) {}

  get isCash(): boolean {
    return this.locationType === 'cash';
  }

  get isBank(): boolean {
    return this.locationType === 'bank';
  }

  get isVault(): boolean {
    return this.locationType === 'vault';
  }
}
